<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns='http://www.w3.org/1999/xhtml' xml:lang='en' lang='en'>
<head>
<title>lightdm.git - lightdm</title>
<meta name='generator' content='cgit v0.10.1'/>
<meta name='robots' content='index, nofollow'/>
<link rel='stylesheet' type='text/css' href='/cgit-data/cgit.css'/>
<link rel='shortcut icon' href='/favicon.ico'/>
<link rel='alternate' title='Atom feed' href='http://pkgs.fedoraproject.org/cgit/lightdm.git/atom/lightdm-1.9.9-FD_leak.patch?h=master' type='application/atom+xml'/>
</head>
<body>
<div id='cgit'><table id='header'>
<tr>
<td class='logo' rowspan='2'><a href='/cgit/'><img src='/cgit-data/cgit.png' alt='cgit logo'/></a></td>
<td class='main'><a href='/cgit/'>index</a> : <a title='lightdm.git' href='/cgit/lightdm.git/'>lightdm.git</a></td><td class='form'><form method='get' action=''>
<select name='h' onchange='this.form.submit();'>
<option value='epel7'>epel7</option>
<option value='f16'>f16</option>
<option value='f17'>f17</option>
<option value='f18'>f18</option>
<option value='f19'>f19</option>
<option value='f20'>f20</option>
<option value='master' selected='selected'>master</option>
</select> <input type='submit' name='' value='switch'/></form></td></tr>
<tr><td class='sub'>lightdm</td><td class='sub right'>Jon Ciesla</td></tr></table>
<table class='tabs'><tr><td>
<a href='/cgit/lightdm.git/'>summary</a><a href='/cgit/lightdm.git/refs/'>refs</a><a href='/cgit/lightdm.git/log/lightdm-1.9.9-FD_leak.patch'>log</a><a class='active' href='/cgit/lightdm.git/tree/lightdm-1.9.9-FD_leak.patch'>tree</a><a href='/cgit/lightdm.git/commit/lightdm-1.9.9-FD_leak.patch'>commit</a><a href='/cgit/lightdm.git/diff/lightdm-1.9.9-FD_leak.patch'>diff</a><a href='/cgit/lightdm.git/stats/lightdm-1.9.9-FD_leak.patch'>stats</a></td><td class='form'><form class='right' method='get' action='/cgit/lightdm.git/log/lightdm-1.9.9-FD_leak.patch'>
<select name='qt'>
<option value='grep'>log msg</option>
<option value='author'>author</option>
<option value='committer'>committer</option>
<option value='range'>range</option>
</select>
<input class='txt' type='text' size='10' name='q' value=''/>
<input type='submit' value='search'/>
</form>
</td></tr></table>
<div class='path'>path: <a href='/cgit/lightdm.git/tree/'>root</a>/<a href='/cgit/lightdm.git/tree/lightdm-1.9.9-FD_leak.patch'>lightdm-1.9.9-FD_leak.patch</a></div><div class='content'>blob: f5abf78ca8f62890e9f85661f51b5643eef06290 (<a href='/cgit/lightdm.git/plain/lightdm-1.9.9-FD_leak.patch'>plain</a>)
<table summary='blob content' class='blob'>
<tr><td class='linenumbers'><pre><a id='n1' href='#n1'>1</a>
<a id='n2' href='#n2'>2</a>
<a id='n3' href='#n3'>3</a>
<a id='n4' href='#n4'>4</a>
<a id='n5' href='#n5'>5</a>
<a id='n6' href='#n6'>6</a>
<a id='n7' href='#n7'>7</a>
<a id='n8' href='#n8'>8</a>
<a id='n9' href='#n9'>9</a>
<a id='n10' href='#n10'>10</a>
<a id='n11' href='#n11'>11</a>
<a id='n12' href='#n12'>12</a>
<a id='n13' href='#n13'>13</a>
<a id='n14' href='#n14'>14</a>
<a id='n15' href='#n15'>15</a>
<a id='n16' href='#n16'>16</a>
<a id='n17' href='#n17'>17</a>
<a id='n18' href='#n18'>18</a>
<a id='n19' href='#n19'>19</a>
<a id='n20' href='#n20'>20</a>
<a id='n21' href='#n21'>21</a>
<a id='n22' href='#n22'>22</a>
<a id='n23' href='#n23'>23</a>
<a id='n24' href='#n24'>24</a>
<a id='n25' href='#n25'>25</a>
<a id='n26' href='#n26'>26</a>
<a id='n27' href='#n27'>27</a>
<a id='n28' href='#n28'>28</a>
<a id='n29' href='#n29'>29</a>
<a id='n30' href='#n30'>30</a>
<a id='n31' href='#n31'>31</a>
</pre></td>
<td class='lines'><pre><code>diff -up lightdm-1.9.9/src/greeter.c.FD_leak lightdm-1.9.9/src/greeter.c
--- lightdm-1.9.9/src/greeter.c.FD_leak	2014-02-23 18:03:55.000000000 -0600
+++ lightdm-1.9.9/src/greeter.c	2014-03-11 08:26:37.667341382 -0500
@@ -969,9 +969,15 @@ greeter_finalize (GObject *object)
         g_object_unref (self-&gt;priv-&gt;authentication_session);
     }
     if (self-&gt;priv-&gt;to_greeter_channel)
+    {
+        close(g_io_channel_unix_get_fd(self-&gt;priv-&gt;to_greeter_channel));
         g_io_channel_unref (self-&gt;priv-&gt;to_greeter_channel);
+    }
     if (self-&gt;priv-&gt;from_greeter_channel)
+    {
+        close(g_io_channel_unix_get_fd(self-&gt;priv-&gt;from_greeter_channel)); 
         g_io_channel_unref (self-&gt;priv-&gt;from_greeter_channel);
+    }
     if (self-&gt;priv-&gt;from_greeter_watch)
         g_source_remove (self-&gt;priv-&gt;from_greeter_watch);
 
diff -up lightdm-1.9.9/src/session.c.FD_leak lightdm-1.9.9/src/session.c
--- lightdm-1.9.9/src/session.c.FD_leak	2014-03-05 21:35:03.000000000 -0600
+++ lightdm-1.9.9/src/session.c	2014-03-11 08:26:37.667341382 -0500
@@ -450,6 +450,8 @@ session_watch_cb (GPid pid, gint status,
     if (session-&gt;priv-&gt;is_guest)
         guest_account_cleanup (session-&gt;priv-&gt;username);
 
+    close(session-&gt;priv-&gt;to_child_input);
+    close(session-&gt;priv-&gt;from_child_output);
     /* Drop our reference on the child process, it has terminated */
     g_object_unref (session);
 }
</code></pre></td></tr></table>
</div> <!-- class=content -->
<div class='footer'>generated  by cgit v0.10.1 at 2014-05-16 21:09:31 (GMT)</div>
</div> <!-- id=cgit -->
</body>
</html>
